enhancements/operators/condition-links.md

```diff
@@ -0,0 +1,206 @@
+---
+title: operator-condition-links
+authors:
+  - "@wking"
+reviewers:
+  - "@abhinavdahiya"
+  - "@LalatenduMohanty"
+approvers:
+  - "@deads2k"
+  - "@sdodson"
+creation-date: 2020-04-24
+last-updated: 2020-04-30
+status: implementable
+---
+
+# Operator Condition Links
+
+When an operator condition is in a surprising state (`Available=False`, `Degraded=True`, etc.), it's not always clear from the short reason and message what the impact or recommended mitigations are.
+By including structured links in the condition, we make it easy for all condition consumers to point to external documentation, where there is more space for detailed discussion.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+Are there restrictions on URI targets?
+Do they all need to point at either docs.openshift.com or Knowledgebase articles?
+Is it ok to also link in-cluster web console pages (e.g. "fix this over here...")?
+GitHub pages?
+kubernetes.io/docs/?
+Other external docs?
+These URIs are baked into release images and need to continue to resolve to their intended target for the supported life of that release, so when there is concern about target stability, we may want to point at an off-cluster redirection service.
+
+Do we require these doc links?
+Some messages may be sufficient without links, in which case creating a doc-link may be busywork.
+On the other hand, even relatively straightforward stuff like [`VersionNotFound`][version-not-found] could benefit from a few lines of context, which is more than we want to bake into the message string itself.
+Deciding this might benefit from a survey of conditions collected in Insights.
```

Comment:
If the issue can be explained without a document URL link that would be better. So I wont suggest making the doc links compulsory.

Reply:
> If the issue can be explained without a document URL link that would be better.

Do you have an example of a surprising condition (Degraded=True, Upgradeable=False) that you think can be explained without a document URI?  Can you paste it in here so we can kick around whether we think it is sufficient on its own?  I'm personally skeptical that we have any URI-less messages that cover the reason the condition is firing, the expected impact of the condition, and suggested mitigations.